---
content_type: page
title: Readings
uid: fe6cb659-ab98-754f-8b3f-78a2e07e97aa
---

Readings are from the course textbook, which has been transcribed and compiled by students in this course over many years. The instructions for student "scribes," and the templates they used, are linked below.

*   [Scribing Guide (PDF)]({{< baseurl >}}/resources/mit6_047f15_scribingguide)
*   [Scribe Templates (ZIP)]({{< baseurl >}}/resources/scribetemplates) (This zip file contains: 1 .sh file and 2 .txt files.)

The entire course textbook is available, courtesy of the professor and the students.

[Kellis, Manolis, ed. _Computational Biology: Genomes, Networks, Evolution. MIT course 6.047 / 6.878_. 2016 (PDF - 43.5MB)](/ans7870/6/6.047/f15/MIT6_047F15_Compiled.pdf).

| LEC # | TOPICS | READINGS |
| --- | --- | --- |
| 1 | Introduction: Course Overview, Biology, Algorithms, Machine Learning |  {{< br >}}{{< br >}} Chapter 1: Introduction to the Course {{< br >}}{{< br >}} *   1.1 Introduction and Goals{{< br >}}*   1.2 Final Project: Introduction to Research in Computational Biology{{< br >}}*   1.3 Additional Materials{{< br >}}*   1.4 Crash Course in Molecular Biology{{< br >}}*   1.5 Introduction to Algorithms and Probabilistic Inference {{< br >}}{{< br >}}  |
| 2 | Alignment I: Dynamic Programming, Global and Local Alignment |  {{< br >}}{{< br >}} Chapter 2: Sequence Alignment and Dynamic Programming {{< br >}}{{< br >}} *   2.1 Introduction{{< br >}}*   2.2 Aligning Sequences{{< br >}}*   2.3 Problem Formulations{{< br >}}*   2.4 Dynamic Programming{{< br >}}*   2.5 The Needleman-Wunsch Algorithm{{< br >}}*   2.6 Multiple Alignment{{< br >}}*   2.7 Current Research Directions{{< br >}}*   2.8 Further Reading{{< br >}}*   2.9 Tools and Techniques{{< br >}}*   2.10 What Have We Learned?{{< br >}}*   2.11 Appendix {{< br >}}{{< br >}}  |
| 3 | Alignment II: Database Search, Rapid String Matching, BLAST, BLOSUM |  {{< br >}}{{< br >}} Chapter 3: Rapid Sequence Alignment and Database Search {{< br >}}{{< br >}} *   3.1 Introduction{{< br >}}*   3.2 Global Alignment vs. Local Alignment vs. Semi-global Alignment{{< br >}}*   3.3 Linear-time Exact String Matching{{< br >}}*   3.4 The BLAST (Basic Local Alignment Search Tool) Algorithm{{< br >}}*   3.5 Pre-processing for Linear-time String Matching{{< br >}}*   3.6 Probabilistic Foundations of Sequence Alignment{{< br >}}*   3.7 Current Research Directions{{< br >}}*   3.8 Further Readings{{< br >}}*   3.9 Tools and Techniques{{< br >}}*   3.10 What Have We Learned? {{< br >}}{{< br >}}  |
| 4 | Hidden Markov Models Part 1: Evaluation / Parsing, Viterbi, Forward Algorithms |  {{< br >}}{{< br >}} Chapter 7: Hidden Markov Models I {{< br >}}{{< br >}} *   7.1 Introduction{{< br >}}*   7.2 Motivation{{< br >}}*   7.3 Markov Chains and HMMS: From Example to Formalizing{{< br >}}*   7.4 Apply HMM to Real World: From Casino to Biology{{< br >}}*   7.5 Algorithmic Settings for HMMs{{< br >}}*   7.6 An Interesting Question: Can We Incorporate Memory in Our Model?{{< br >}}*   7.7 Further Reading{{< br >}}*   7.8 Current Research Directions{{< br >}}*   7.9 Tools and Techniques{{< br >}}*   7.10 What Have We Learned? {{< br >}}{{< br >}}  |
| 5 | Hidden Markov Models Part 2: Posterior Decoding, Learning, Baum-Welch |  {{< br >}}{{< br >}} Chapter 8: Hidden Markov Models II-Posterior Decoding and Learning {{< br >}}{{< br >}} *   8.1 Review of Previous Lecture{{< br >}}*   8.2 Posterior Decoding{{< br >}}*   8.3 Encoding Memory in an HMM: Detection of CpG Islands{{< br >}}*   8.4 Learning{{< br >}}*   8.5 Using HMMs to Align Sequences with Affine Gap Penalties{{< br >}}*   8.6 Current Research Directions{{< br >}}*   8.7 Further Reading{{< br >}}*   8.8 Tools and Techniques{{< br >}}*   8.9 What Have We Learned? {{< br >}}{{< br >}}  |
| 6 | Transcript Structure: GENSCAN, RNA-seq, Mapping, De Novo Assembly, Diff Expr |  {{< br >}}{{< br >}} Chapter 12: Large Intergenic Non-coding RNAs {{< br >}}{{< br >}} *   12.3 Practical Topic: RNAseq {{< br >}}{{< br >}}  |
| 7 | Expression Analysis: Clustering / Classification, K-Means, Hierarchical, Bayesian |  {{< br >}}{{< br >}} Chapter 15: Gene Regulation 1: Gene Expression Clustering {{< br >}}{{< br >}} *   15.1 Introduction{{< br >}}*   15.2 Methods for Measuring Gene Expression{{< br >}}*   15.3 Clustering Algorithms{{< br >}}*   15.4 Current Research Directions{{< br >}}*   15.5 Further Reading{{< br >}}*   15.6 Resources{{< br >}}*   15.7 What Have We Learned?  {{< br >}}{{< br >}} Chapter 16: Gene Regulation 2: Classification {{< br >}}{{< br >}} *   16.1 Introduction{{< br >}}*   16.2 Classification - Bayesian Techniques{{< br >}}*   16.3 Classification Support Vector Machines{{< br >}}*   16.4 Tumor Classification with SVMs{{< br >}}*   16.5 Semi-Supervised Learning{{< br >}}*   16.6 Current Research Directions{{< br >}}*   16.7 Further Reading{{< br >}}*   16.8 Resources {{< br >}}{{< br >}}  |
| 8 | Networks I: Bayesian Inference, Deep Learning, Network Dynamics |  {{< br >}}{{< br >}} Chapter 20: Networks I Inference, Structure, Spectral Methods {{< br >}}{{< br >}} *   20.1 Introduction{{< br >}}*   20.2 Network Centrality Measures{{< br >}}*   20.3 Linear Algebra Review{{< br >}}*   20.4 Sparse Principal Component Analysis{{< br >}}*   20.5 Network Communities and Modules{{< br >}}*   20.6 Network Diffusion Kernels{{< br >}}*   20.7 Neural Networks{{< br >}}*   20.8 Open Issues and Challenges{{< br >}}*   20.9 Current Research Directions{{< br >}}*   20.10 Further Reading{{< br >}}*   20.11 Tools and Techniques{{< br >}}*   20.12 What Have We Learned? {{< br >}}{{< br >}} Chapter 21: Regulatory Networks: Inferences, Analysis, Application {{< br >}}{{< br >}} *   21.1 Introduction{{< br >}}*   21.2 Structure Inference{{< br >}}*   21.3 Overview of the OGM Learning Task{{< br >}}*   21.4 Applications of Networks{{< br >}}*   21.5 Structural Properties of Networks{{< br >}}*   21.6 Network Clustering {{< br >}}{{< br >}}  |
| 9 | Networks II: Network Learning, Structure, Spectral Methods |
| 10 | Regulatory Motifs: Discovery, Representation, PBMs, Gibbs Sampling, EM |  {{< br >}}{{< br >}} Chapter 17: Regulatory Motifs, Gibbs Sampling, and EM {{< br >}}{{< br >}} *   17.1 Introduction to Regulatory Motifs and Gene Regulation{{< br >}}*   17.2 Expectation Maximization{{< br >}}*   17.3 Gibbs Sampling: Sample from Joint (M, Zjj) Distribution{{< br >}}*   17.4 De Novo Motif Discovery{{< br >}}*   17.5 Evolutionary Signatures for Instance Identification{{< br >}}*   17.6 Phylogenies, Branch Length Score, Confidence Score{{< br >}}*   17.7 Possibly Deprecated Stuff Below{{< br >}}*   17.8 Comparing Different Methods{{< br >}}*   17.9 OOPS, ZOOPS, TCM{{< br >}}*   17.10 Extension of the EM Approach{{< br >}}*   17.11 Motif Representation and Information Content {{< br >}}{{< br >}}  |
| 11 | Epigenomics: ChIP-Seq, Read Mapping, Peak Calling, IDR, Chromatin States |  {{< br >}}{{< br >}} Chapter 19: Epigenomics / Chromatin States {{< br >}}{{< br >}} *   19.1 Introduction{{< br >}}*   19.2 Epigenetic Information in Nucleosomes{{< br >}}*   19.3 Epigenomic Assays{{< br >}}*   19.4 Primary Data Processing of ChIP Data{{< br >}}*   19.5 Annotating the Genome Using Chromatin Signatures{{< br >}}*   19.6 Current Research Directions{{< br >}}*   19.7 Further Reading{{< br >}}*   19.8 Tools and Techniques{{< br >}}*   19.9 What Have We Learned? {{< br >}}{{< br >}}  |
| 12 | RNA Modifications: RNA Editing, Translation Regulation, Splicing Regulation |  {{< br >}}{{< br >}} Chapter 11: RNA Modifications {{< br >}}{{< br >}} *   11.1 Introduction{{< br >}}*   11.2 Post-transcriptional Regulation{{< br >}}*   11.3 Current Research Directions{{< br >}}*   11.4 Further Reading{{< br >}}*   11.5 Tools and Techniques{{< br >}}*   11.6 What Have We Learned? {{< br >}}{{< br >}}  |
| 13 | Resolving Human Ancestry and Human History from Genetic Data |  {{< br >}}{{< br >}} Chapter 29: Population History {{< br >}}{{< br >}} *   29.1 Introduction{{< br >}}*   29.2 Quick Survey of Human Genetic Variation{{< br >}}*   29.3 African European Gene Flow{{< br >}}*   29.4 Gene Flow on the Indian Subcontinent{{< br >}}*   29.5 Gene Flow Between Archaic Human Populations{{< br >}}*   29.6 European Ancestry and Migrations{{< br >}}*   29.7 Tools and Techniques{{< br >}}*   29.8 Further Directions{{< br >}}*   29.9 Further Reading {{< br >}}{{< br >}}  |
| 14 | Disease Association Mapping, GWAS, Organismal Phenotypes |  {{< br >}}{{< br >}} Chapter 31: Medical Genetics-The Past to the Present {{< br >}}{{< br >}} *   31.1 Introduction{{< br >}}*   31.2 Goals of Investigating the Genetic Basis of Disease{{< br >}}*   31.3 Mendelian Traits{{< br >}}*   31.4 Complex Traits{{< br >}}*   31.5 Genome-wide Association Studies{{< br >}}*   31.6 Current Research Directions{{< br >}}*   31.7 Further Reading{{< br >}}*   31.8 Tools and Techniques{{< br >}}*   31.9 What Have We Learned? {{< br >}}{{< br >}}  |
| 15 | Quantitative Trait Mapping, Molecular Traits, eQTLs |  {{< br >}}{{< br >}} Chapter 32: Variation 2: Quantitative Trait Mapping, eQTLs, Molecular Trait Variation {{< br >}}{{< br >}} *   32.1 Introduction{{< br >}}*   32.2 eQTL Basics{{< br >}}*   32.3 Structure of an eQTL Study{{< br >}}*   32.4 Current Research Directions{{< br >}}*   32.5 What Have We Learned?{{< br >}}*   32.6 Further Reading{{< br >}}*   32.7 Tools and Resources {{< br >}}{{< br >}}  |
| 16 | Missing Heritability, Complex Traits, Interpret GWAS, Rank-based Enrichment |  {{< br >}}{{< br >}} Chapter 33: Missing Heritability {{< br >}}{{< br >}} *   33.1 Introduction{{< br >}}*   33.2 Current Research Directions{{< br >}}*   33.3 Further Reading{{< br >}}*   33.4 Tools and Techniques{{< br >}}*   33.5 What Have We Learned? {{< br >}}{{< br >}}  |
| 17 | Comparative Genomics and Evolutionary Signatures |  {{< br >}}{{< br >}} Chapter 4: Comparative Genomics I: Genome Annotation {{< br >}}{{< br >}} *   4.1 Introduction{{< br >}}*   4.2 Conservation of Genomic Sequences{{< br >}}*   4.3 Excess Constraint{{< br >}}*   4.4 Diversity of Evolutionary Signatures: An Overview of Selection Patterns{{< br >}}*   4.5 Protein-coding Signatures{{< br >}}*   4.6 MicroRNA (miRNA) Gene Signatures{{< br >}}*   4.7 Regulatory Motifs{{< br >}}*   4.8 Current Research Directions{{< br >}}*   4.9 Further Reading{{< br >}}*   4.10 Tools and Techniques{{< br >}}*   4.11 Bibliography {{< br >}}{{< br >}}  |
| 18 | Phylogenetics: Molecular Evolution, Tree Building, Phylogenetic Inference |  {{< br >}}{{< br >}} Chapter 27: Molecular Evolution and Phylogenetics {{< br >}}{{< br >}} *   27.1 Introduction{{< br >}}*   27.2 Basics of Phylogeny{{< br >}}*   27.3 Distance Based Methods{{< br >}}*   27.4 Character Based Methods{{< br >}}*   27.5 Possible Theoretical and Practical Issues with Discussed Approach{{< br >}}*   27.6 Towards Final Project{{< br >}}*   27.7 What Have We Learned? {{< br >}}{{< br >}}  |
| 19 | Phylogenomics: Gene / Species Trees, Reconciliation, Recombination Graphs |  {{< br >}}{{< br >}} Chapter 28: Phylogenomics II {{< br >}}{{< br >}} *   28.1 Introduction{{< br >}}*   28.2 Inferring Orthologs / Paralogs, Gene Duplication and Loss{{< br >}}*   28.3 Reconstruction{{< br >}}*   28.4 Modeling Population and Allele Frequencies{{< br >}}*   28.5 SPIDIR{{< br >}}*   28.6 Ancestral Recombination Graphs{{< br >}}*   28.7 Conclusion{{< br >}}*   28.8 Current Research Directions{{< br >}}*   28.9 Further Reading{{< br >}}*   28.10 Tools and Techniques{{< br >}}*   28.11 What Have We Learned? {{< br >}}{{< br >}}  |
| 20 | Personal Genomics, Disease Epigenomics: Systems Approaches to Disease |  {{< br >}}{{< br >}} Chapter 34: Personal Genomes, Synthetic Genomes, Computing in C vs. Si {{< br >}}{{< br >}} *   34.1 Introduction{{< br >}}*   34.2 Reading and Writing Genomes{{< br >}}*   34.3 Personal Genomes{{< br >}}*   34.4 Current Research Directions{{< br >}}*   34.5 Further Reading{{< br >}}*   34.6 Tools and Techniques{{< br >}}*   34.7 What Have We Learned? {{< br >}}{{< br >}} Chapter 36: Cancer Genomics {{< br >}}{{< br >}} *   36.1 Introduction{{< br >}}*   36.2 Characterization{{< br >}}*   36.3 Interpretation{{< br >}}*   36.4 Current Research Directions{{< br >}}*   36.5 Further Reading{{< br >}}*   36.6 Tools and Techniques{{< br >}}*   36.7 What Have We Learned? {{< br >}}{{< br >}}  |
| 21 | Three-Dimensional Chromatin Interactions: 3C, 5C, HiC, ChIA-Pet |  {{< br >}}{{< br >}} Chapter 30: Population Genetic Variation {{< br >}}{{< br >}} *   30.1 Introduction{{< br >}}*   30.2 Population Selection Basics{{< br >}}*   30.3 Genetic Linkage{{< br >}}*   30.4 Natural Selection{{< br >}}*   30.5 Human Evolution{{< br >}}*   30.6 Current Research{{< br >}}*   30.7 Further Reading {{< br >}}{{< br >}}  |
| 22 | Genome Engineering with CRISPR / Cas9 and Related Technologies | No readings for this lecture.